pkg/console/operator/operator.go

```diff
@@ -176,38 +176,53 @@ func (c *consoleOperator) Sync(obj metav1.Object) error {
 		return err
 	}
 
-	// all configs needed to do a sync
 	if err := c.handleSync(operatorConfig, consoleConfig, infrastructureConfig); err != nil {
-		c.SyncStatus(c.ConditionFailing(operatorConfig, "SyncLoopError", "Operator sync loop failed to complete."))
 		return err
 	}
-	c.SyncStatus(c.ConditionNotFailing(operatorConfig))
+
 	return nil
 }
 
-func (c *consoleOperator) handleSync(originalOperatorConfig *operatorsv1.Console, consoleConfig *configv1.Console, infrastructureConfig *configv1.Infrastructure) error {
-	operatorConfig := originalOperatorConfig.DeepCopy()
+func (c *consoleOperator) handleSync(operatorConfig *operatorsv1.Console, consoleConfig *configv1.Console, infrastructureConfig *configv1.Infrastructure) error {
+	operatorConfigCopy := operatorConfig.DeepCopy()
 
-	switch operatorConfig.Spec.ManagementState {
+	switch operatorConfigCopy.Spec.ManagementState {
 	case operatorsv1.Managed:
 		logrus.Println("console is in a managed state.")
 		// handled below
 	case operatorsv1.Unmanaged:
 		logrus.Println("console is in an unmanaged state.")
-		c.SyncStatus(c.ConditionsManagementStateUnmanaged(operatorConfig))
+		c.SyncStatus(c.ConditionsManagementStateUnmanaged(operatorConfigCopy))
 		return nil
 	case operatorsv1.Removed:
 		logrus.Println("console has been removed.")
-		c.SyncStatus(c.ConditionsManagementStateRemoved(operatorConfig))
-		return c.deleteAllResources(operatorConfig)
+		c.SyncStatus(c.ConditionsManagementStateRemoved(operatorConfigCopy))
+		return c.deleteAllResources(operatorConfigCopy)
 	default:
-		// TODO should update status
-		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+		c.SyncStatus(c.ConditionsManagementStateInvalid(operatorConfigCopy))
+		return fmt.Errorf("console is in an unknown state: %v", operatorConfigCopy.Spec.ManagementState)
 	}
 
-	_, _, _, err := sync_v400(c, operatorConfig, consoleConfig, infrastructureConfig)
+	// we can default to not failing, and wait to see if sync returns an error
+	c.ConditionNotFailing(operatorConfigCopy)
+	_, _, _, err := sync_v400(c, operatorConfigCopy, consoleConfig, infrastructureConfig)
```

Comment:
Not new, but do we need the extra return values in `sync_v400` if we just ignore them?

Reply:
We don't, I'll remove them.  These values assumed we would do operator status here (before we knew what that would look like).